### PR TITLE
WFLY-5470 Update subsystem archetype and example to WildFly 10.0.0.CR2

### PR DESCRIPTION
**Pull request by mkjensen**
Update wildfly-subsystem-archetype and wildfly-subsystem-example to use WildFly 10.0.0.CR2.

### CODE REVIEW

File: wildfly-subsystem-archetype/pom.xml [hunk 1]
```diff
@@ -145,27 +144,11 @@
...
-                <scope>test</scope>
-                <version>${version.wildfly.core}</version>
-            </dependency>
-            <dependency>
-                <groupId>junit</groupId>
-                <artifactId>junit</artifactId>
-                <scope>test</scope>
-                <version>${version.junit}</version>
+                <version>${version.wildfly}</version>
+                <scope>import</scope>
```
Comment (ctomc) [inline wildfly-subsystem-archetype/pom.xml]: never ever do this. You don't want all wildfly core dependencies to be part of your build.
This is huge overkill.

File: wildfly-subsystem-example/pom.xml [hunk 2]
```diff
@@ -130,27 +129,11 @@
...
-                <scope>test</scope>
-                <version>${version.wildfly.core}</version>
-            </dependency>
-            <dependency>
-                <groupId>junit</groupId>
-                <artifactId>junit</artifactId>
-                <scope>test</scope>
-                <version>${version.junit}</version>
+                <version>${version.wildfly}</version>
+                <scope>import</scope>
```
Comment (ctomc) [inline wildfly-subsystem-example/pom.xml]: same here

File: wildfly-subsystem-archetype/pom.xml (hunk 1 above)
Comment (mkjensen) [inline wildfly-subsystem-archetype/pom.xml]: Okay. I just wanted to include the dependency management part of wildfly-parent to get the versions used there for wildfly-core etc., making it easy to just set version.wildfly and then get everything aligned.

Comment (ctomc) [inline wildfly-subsystem-archetype/pom.xml]: sure, but with this you get almost 300 components as dependency which bit huge.
In most cases even having all wildfly-core deps would be too much, but would make more sense.

Comment (mkjensen) [inline wildfly-subsystem-archetype/pom.xml]: I see. I have pushed a new commit that does not use the import approach.

**Comment by ctomc** [comment]
Thank you

**Pull request merged**